app/serializers/user_project_preference_serializer.rb

```diff
@@ -16,8 +16,15 @@ def activity_count
     end
   end
 
+  def activity_count_by_workflow
+    UserSeenSubject.activity_by_workflow(@model.user_id, project_workflows_ids)
+  end
+
   def user_project_activity
-    project_workflows_ids = Workflow.where(project_id: @model.project_id).pluck(:id)
     UserSeenSubject.count_user_activity(@model.user_id, project_workflows_ids)
   end
+
+  def project_workflows_ids
+    Workflow.where(project_id: @model.project_id).pluck(:id)
```

Comment:
should we cache this call? the AR call will probably be cached but we are using it possibly twice per serialized UPP.